library/src/test/java/com/novoda/downloadmanager/lib/BatchRepositoryTest.java

```diff
@@ -53,16 +85,17 @@ private Cursor singleBatchCursor(final long batchId) {
         int idColumn = 1337;
         Cursor cursor = mock(Cursor.class);
         when(cursor.getColumnIndexOrThrow(Downloads.Impl.Batches._ID)).thenReturn(idColumn);
-        when(cursor.moveToNext()).thenAnswer(new Answer<Boolean>() {
-            int count = 0;
-
-            @Override
-            public Boolean answer(InvocationOnMock invocation) throws Throwable {
-                boolean batchSizeNotReached = count < batchSize;
-                count++;
-                return batchSizeNotReached;
-            }
-        });
+        when(cursor.moveToNext()).thenAnswer(
```

Comment:
why do you have to do this? Is there are reason you can't use the indexed cursor below?